api/python/provisioner/commands/check.py

```diff
@@ -0,0 +1,392 @@
+    """
+
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = CheckArgs
+
+    _PRV_METHOD_MOD = "_"  # private method modificator
+    # -c 1 means count of sending ECHO_REQUEST packets
+    # -W 1 means timeout to wait for a response
+    _PING_CMD_TMP = "ping -c 1 -W 1 {server_addr}"  # ping command template
+    _BMC_CHECK_CMD = 'ipmitool chassis status | grep "System Power"'
+    _CLUSTER_STATUS_CHECK_CMD = "pcs status --full cluster"
+    _LOGS_ARE_GOOD_CHECK_CMD_TMP = ('test -f {log_filename} && '
+                                    'grep -i "{key_phrase}" {log_filename} '
+                                    '1>/dev/null && exit 1')
+    _PSWDLESS_SSH_CHECK_CMD_TMP = ('ssh -o BatchMode=yes '
+                                   '-o PasswordAuthentication=no '
+                                   '{user}@{hostname} exit &>/dev/null '
+                                   '&& test $? == 0 || exit 1')
+
+    @staticmethod
+    def _network(*, args: str, targets: str, dry_run: bool = False) -> dict:
+        """
+        Private method for network checks.
+
+        :param args: network specific checking parameters and arguments
+        :param targets: target nodes where network checks will be executed
+        :param bool dry_run: for debugging purposes. Execute method without
+                             real command execution on target nodes
+        :return:
+        """
+        if dry_run:
+            return dict()
+
+        res = dict()
+
+        minion_id = local_minion_id()
+
+        cluster_path = KeyPath('cluster')
+        node_network_path = KeyPath(f'cluster/{minion_id}/network/data_nw')
+
+        pillar_keys = (PillarKey(cluster_path / 'type'),
+                       PillarKey(node_network_path / "public_ip_addr"),
+                       PillarKey(node_network_path / "pvt_ip_addr"),
+                       PillarKey(cluster_path / "cluster_ip"))
+
+        # maybe we need to call it from the all targets and analyze the output
+        # from each node
+        pillar = PillarResolver(cfg.LOCAL_MINION).get(pillar_keys)
+
+        pillar = pillar.get(local_minion_id())  # type: dict
+
+        for key in pillar_keys:
+            if not pillar[key] or pillar[key] is values.MISSED:
+                res[str(key.keypath)] = (f'{cfg.CheckVerdict.FAIL.value}: '
+                                         'value is not specified')
+            elif pillar[key] is None:
+                res[str(key.keypath)] = (f'{cfg.CheckVerdict.FAIL.value}: '
+                                         'value is unset')
+            else:
+                res[str(key.keypath)] = cfg.CheckVerdict.PASSED.value
+
+        return res
+
+    @staticmethod
+    def _communicability(*, args: str, targets: str = cfg.ALL_MINIONS,
+                         dry_run: bool = False) -> dict:
+        """
+        Check if nodes can communicate using `salt test.ping`
```

Comment:
Test.ping verifies  master minion connection.please adjust the title a bit. Also test.ping is not the best choice since it will hang if a minion is down. Consider to use that call https://github.com/Seagate/cortx-prvsnr/blob/9e5677a67a14480acc3f7690f33e003fee3a346f/api/python/provisioner/salt_minion.py#L49 instead.

Reply:
Done